parser.y

```diff
@@ -1586,9 +1587,13 @@ AlterTablePartitionOpt:
 	PartitionOpt
 	{
 		if $1 != nil {
-			$$ = &ast.AlterTableSpec{
-				Tp:        ast.AlterTablePartition,
-				Partition: $1.(*ast.PartitionOptions),
+			if alterTableSpec, ok := $1.(*ast.AlterTableSpec); ok {
```

Comment:
I suggest adding a new term `ATTRIBUTES : "ATTRIBUTES" "=" stringLit | "DEFAULT"`, which will return the attributes spec. And the change here is not elegant for me. It is strange if `PartitionOpt` did not return `PartitionOptions`.